### PR TITLE
Agent: Refactor _check_tcp_ports to resolve C901

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #1076 .

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [x] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/infection_monkey/network_scanning/tcp_scanner.py [hunk 1]
```diff
@@ -94,89 +94,105 @@ def _check_tcp_ports(
...
+            possible_ports_sockets.add((port, sock))
+        elif err == 10035:  # WSAEWOULDBLOCK is valid.
+            # https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-connect
+            # says, "Use the select function to determine the completion of the connection
+            # request by checking to see if the socket is writable," which is being done below.
+            possible_ports_sockets.add((port, sock))
+        elif err == 115:  # EINPROGRESS     115     /* Operation now in progress */
+            possible_ports_sockets.add((port, sock))
+        else:
+            logger.warning("Failed to connect to port %s, error code is %d", port, err)
```
Comment (shreyamalviya) [inline monkey/infection_monkey/network_scanning/tcp_scanner.py:154]: Doesn't this raise C901 again?

Comment (ilija-lazoroski) [reply monkey/infection_monkey/network_scanning/tcp_scanner.py:154]: Didn't see one...

**Comment by ilija-lazoroski** [comment]
EtE passed!

**Review by mssalvatore — APPROVED** [review]
I think there's still a lot of room to improve this code. Our goal here is to resolve C901 warnings and this achieves that goal. Approved pending testing.

File: monkey/infection_monkey/network_scanning/tcp_scanner.py [hunk 2]
```diff
@@ -94,89 +94,105 @@ def _check_tcp_ports(
...
     """
-    sockets = [socket.socket(socket.AF_INET, socket.SOCK_STREAM) for _ in range(len(ports_to_scan))]
-    for s in sockets:
-        s.setblocking(False)
-
-    possible_ports = set()
-    connected_ports = set()
-    open_ports = {}
+    ports_sockets = _create_sockets(ports_to_scan)
+    possible_ports, connected_ports_sockets = _connect_sockets(ip, ports_sockets)
```
Comment (mssalvatore) [inline monkey/infection_monkey/network_scanning/tcp_scanner.py:99]: Does this need to bewrapped in try/ercept? It was in the past.

**Pull request merged**